test/unit/builders/test_violators_avaliations_builder.py

```diff
@@ -1,115 +1,324 @@
+                license_plate=LicensePlate('QBJ-6840'),
+                type_infraction=TypeInfractionStrings.AVERAGE,
+                infraction_date=datetime(1890, 1, 1, 12),
+                notification_date=datetime(1890, 1, 2, 7)
+            ),
+            TrafficViolation(
+                identity_card=IdentityCard('13.386.966-0', 'Miho'),
+                license_plate=LicensePlate('RXO-0694'),
+                type_infraction=TypeInfractionStrings.SERIOUS,
+                infraction_date=datetime(2000, 6, 1, 12),
+                notification_date=datetime(2000, 6, 12, 7)
+            ),
+            TrafficViolation(
+                identity_card=IdentityCard('19.632.142-6', 'Takashi'),
+                license_plate=LicensePlate('IDE-3516'),
+                type_infraction=TypeInfractionStrings.VERY_SERIOUS,
+                infraction_date=datetime(2000, 2, 2, 12),
+                notification_date=datetime(2000, 2, 3, 7)
+            )
+        ]
+        violators_avaliations_builder = ViolatorsAvaliationsBuilder(
+            traffic_violations
+        )
+
+        violators_avaliations = (
+            violators_avaliations_builder.build_violators_avaliations()
+        )
+
+        self.assertEqual(
+            violators_avaliations, [
+                ViolatorAvaliation(
+                    identity_card=IdentityCard('13.846.994-5', 'Morgan'),
+                    license_plates=['MGN-9130'],
+                    demerit_points=3
+                ),
+                ViolatorAvaliation(
+                    identity_card=IdentityCard('37.594.403-5', 'Det. Olivera'),
+                    license_plates=['QBJ-6840'],
+                    demerit_points=4
+                ),
+                ViolatorAvaliation(
+                    identity_card=IdentityCard('13.386.966-0', 'Miho'),
+                    license_plates=['RXO-0694'],
+                    demerit_points=5
+                ),
+                ViolatorAvaliation(
+                    identity_card=IdentityCard('19.632.142-6', 'Takashi'),
+                    license_plates=['IDE-3516'],
+                    demerit_points=7
+                )
+            ]
+        )
+
+    def test_build_violator_avaliation_invalid_demerit_points(self):
+        traffic_violations = [
+            TrafficViolation(
+                identity_card=IdentityCard('13.846.994-5', 'Morgan'),
+                license_plate=LicensePlate('MGN-9130'),
+                type_infraction=TypeInfractionStrings.VERY_SERIOUS,
+                infraction_date=datetime(1750, 1, 1, 12),
+                notification_date=datetime(1750, 5, 4, 7)
+            ),
+            TrafficViolation(
+                identity_card=IdentityCard('37.594.403-5', 'Det. Olivera'),
+                license_plate=LicensePlate('QBJ-6840'),
+                type_infraction=TypeInfractionStrings.AVERAGE,
+                infraction_date=datetime(1890, 1, 1, 12),
```

Comment:
As datas ficaram bem longe dos 30 dias, seria legal vc garantir que a regra ta sendo aplicada de forma correta. E se for 30 dias, e se for 29 dias e se for 31 dias o que acontece?